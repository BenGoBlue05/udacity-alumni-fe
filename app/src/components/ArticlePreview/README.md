## ArticlePreview Component
A component that shows a preview of an article.

### Example

```js
<ArticlePreview article={article} />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **article**    | Object   |             | An object value, with the keys shown below

Article Keys:
user: {
  name: PropTypes.string.isRequired,
},
feature_image: PropTypes.string.isRequired,
content: PropTypes.string.isRequired,
id: PropTypes.number.isRequired,


### Other Information
Reused on search and landing pages as of 10/8/2016
