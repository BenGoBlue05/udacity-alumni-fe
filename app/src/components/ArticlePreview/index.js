import React, { PropTypes } from 'react';
import Anchor from 'grommet-udacity/components/Anchor';
import Box from 'grommet-udacity/components/Box';
import Card from 'grommet-udacity/components/Card';


const ArticlePreview = ({
  article,
}) => (
  <Box size="medium" pad={{ vertical: 'medium' }}>
    <Card
      label={`By ${article.user.name}`}
      thumbnail={article.feature_image}
      heading={article.title.slice(0, 15)}
      description={`${article.content.slice(0, 200)}...`}
      link={
        <Anchor
          href={`/articles/${article.id}`}
          primary
          label="Read More"
        />
      }
    />
  </Box>
);

ArticlePreview.propTypes = {
  article: PropTypes.shape({
    user: PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
    feature_image: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  }),
};

export default ArticlePreview;
