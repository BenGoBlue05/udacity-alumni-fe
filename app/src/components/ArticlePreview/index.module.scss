.articlePreview {
  p {
    font-family: 'Lato';
    font-size: 18px !important;
    line-height: 1.58 !important;
    letter-spacing: -.004em !important;
    font-weight: 400 !important;
  }
  h2 {
    margin-top: 3px;
    font-size: 29px;
    margin-left: -1.81px;
    line-height: 1.04;
    letter-spacing: -.028em;
  }
}
