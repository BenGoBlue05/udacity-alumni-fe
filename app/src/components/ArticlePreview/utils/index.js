/* eslint-disable */
// helper to insert mark tags
function insertMarkTag(sourceString, subString) {
  return sourceString.replace(new RegExp("(" + subString + ")", "g"), "<mark>$1</mark>");
}

// {`${article.content.slice(0, 200)}...`}
export const highlightContent = (searchTerm, text) => {
  if (text.indexOf(searchTerm) !== -1) {
    /**
      TODO: do a better algorythm to get the right text section containing
      the search term. Right now we only show the first 200 characters of the text.
      This will be a problem for bigger texts.
    **/
    const formattedText = insertMarkTag(text, searchTerm);
    return formattedText.slice(0, 200);
  }
  else {
    return text.slice(0, 200);
  }
};
/* eslint-enable */
