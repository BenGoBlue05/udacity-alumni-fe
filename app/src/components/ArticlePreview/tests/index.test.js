import ArticlePreview from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<ArticlePreview />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <ArticlePreview />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
