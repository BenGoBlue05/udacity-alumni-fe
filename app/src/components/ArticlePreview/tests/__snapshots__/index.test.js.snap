exports[`<ArticlePreview /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad={
    Object {
      "vertical": "medium"
    }
  }
  responsive={true}
  size="medium"
  tag="div">
  <Card
    contentPad="medium"
    description="Diagnostic related group HIPAA accreditation"
    heading="Current procedu"
    headingStrong={true}
    label="By Andreas Daimainger"
    link={
      <Anchor
        href="/articles/current-procedural-terminology-benefit-impaired-risk"
        label="Read More"
        primary={true}
        tag="a" />
    }
    textSize="small"
    thumbnail={
      <ResponsiveImage
        matchHeight={true}
        src="https://robohash.org/sapienteeaoptio.png?size=300x300" />
    } />
</Box>
`;
